platform/security/common/src/test/java/org/codice/ddf/security/common/SPMetadataParserTest.java

```diff
@@ -0,0 +1,107 @@
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.security.common;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+import com.google.common.collect.ImmutableSet;
+import ddf.security.samlp.SamlProtocol.Binding;
+import ddf.security.samlp.impl.EntityInformation;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class SPMetadataParserTest {
+  private static List<String> spMetadata;
+
+  private static Set<Binding> bindingSet;
+
+  private static final Map EMPTY_MAP = Collections.emptyMap();
+
+  private static final String FILE_PATH = "src/test/resources/SPMetadata";
+
+  private static final String ENTITY_ID = "https://localhost:8993/services/saml";
+  private static final String LOGOUT_URL = "https://localhost:8993/services/saml/logout";
+  private static final String ACS_URL = "https://localhost:8993/services/saml/sso";
+  private static final String CERTIFICATE =
+      "MIICsjCCAhugAwIBAgIGAWBQwru6MA0GCSqGSIb3DQEBCwUAMIGEMQswCQYDVQQGEwJVUzELMAkGA1UECBMCQVoxDDAKBgNVBAoTA0RERjEMMAoGA1UECxMDRGV2MRkwFwYDVQQDExBEREYgRGVtbyBSb290IENBMTEwLwYJKoZIhvcNAQkBFiJlbWFpbEFkZHJlc3M9ZGRmcm9vdGNhQGV4YW1wbGUub3JnMCAXDTE3MTIxMjE2NDM0N1oYDzIxMTcxMjEyMTY0MzQ3WjAUMRIwEAYDVQQDDAlsb2NhbGhvc3QwggEiMA0GCSqGSIb3DQEBAQUAA4IBDwAwggEKAoIBAQDEwOqpKuMr+GmiT3kKcMKp4361ByVBaxvj5M6rptn+Al8oOZfxfaDOhOikUVuaPTuIYvTWW04KaQNHI3nm86RwldbwMlPZka5jGeYkOJ3qah1C6OMmMDEC7lM0/bhxPLc8C0cUwAG8FdlBLDKCYFjHj6UBq5xhD/1uJyvSoWcFqJQbXP+igVV7tDdRxRDhp8f5eActecjQXK5+bpzNs8KC+3gvnhBj/ux/DYyrE6iA4nKMnelIaZjhW11bsuQwM6XK8Tq7PQ2LuAuhFoKUq/p8p8Sva75/3nGeCHOlcjj7rnoSCcUgX+u1nnCN/wsshyRAX30wS7C75nQilQkxCfAlAgMBAAGjGDAWMBQGA1UdEQQNMAuCCWxvY2FsaG9zdDANBgkqhkiG9w0BAQsFAAOBgQAQBBked2MsqlmIwxSUXLee6f9R14GPsNsOkLohFhBTKuxC++n+q0+LvMRLmFxPj4dois/dHKQiD7wYWYY4WPvb3nD2aPev4NZm8erB9+XPVueRzxaLXQ+K1oW9Kjui8+2q1lTBBGOjma1DT/b6FVhyJo5SjQ3JMKe9TACVp/YA7A==";
+
+  @BeforeClass
+  public static void setupClass() throws Exception {
+
+    // read SPMetadata file into spMetadata
+    spMetadata = new ArrayList<>();
+    try (Stream<String> lines = Files.lines(Paths.get(FILE_PATH))) {
+      spMetadata.add(lines.collect(Collectors.joining(System.lineSeparator())));
+    }
+
+    // set up binding set
+    bindingSet = ImmutableSet.of(Binding.HTTP_POST, Binding.HTTP_REDIRECT, Binding.SOAP);
+  }
+
+  @Test
+  public void parseNullMetadata() throws Exception {
+    Map<String, EntityInformation> testMap = SPMetadataParser.parse(null, bindingSet);
+
+    assertThat("Returns an empty map", testMap, is(equalTo(EMPTY_MAP)));
```

Comment:
✏️  I'm in favor of using `assertEquals("Returns an empty map", EMPTY_MAP, testMap);` instead of doing `is(equalTo(EMPTY_MAP))`

Reply:
For the sake of our "standards" I'm going to keep the assertions this way.